src/app/loadout-builder/utils.ts

```diff
@@ -369,56 +365,19 @@ function generateMixesFromPerksOrStats(
   return mixes;
 }
 
-export function getBaseStatValues(
-  item: DimItem,
-  assumeMasterwork: boolean | null,
-  lockedModStats?: { [statHash: number]: number }
-) {
+/**
+ * This gets stat values for an item ordered by the statValues array.
+ */
+function getOrderedStatValues(item: DimItem) {
   const stats = _.keyBy(item.stats, (stat) => stat.statHash);
-  const baseStats = {};
+  const keyedStats = {};
 
   for (const statHash of statValues) {
-    baseStats[statHash] = stats[statHash]?.value || 0;
+    keyedStats[statHash] = stats[statHash]?.value || 0;
   }
 
-  // Checking energy tells us if it is Armour 2.0
-  if (item.isDestiny2() && item.sockets && item.energy) {
-    let masterworkSocketHashes: number[] = [];
-
-    // only get masterwork sockets if we aren't manually adding the values
-    if (!assumeMasterwork) {
-      masterworkSocketHashes = getMasterworkSocketHashes(
-        item.sockets,
-        DestinySocketCategoryStyle.EnergyMeter
-      );
-    }
-
-    for (const socket of item.sockets.sockets) {
-      const plugHash = socket?.plug?.plugItem?.hash ?? NaN;
-
-      if (socket.plug?.stats && !masterworkSocketHashes.includes(plugHash)) {
-        for (const statHash of statValues) {
-          if (socket.plug.stats[statHash]) {
-            baseStats[statHash] -= socket.plug.stats[statHash];
-          }
-        }
-      }
-    }
-
-    if (assumeMasterwork) {
-      for (const statHash of statValues) {
-        baseStats[statHash] += 2;
-      }
-    }
-    // For Armor 2.0 mods, include the stat values of any locked mods in the item's stats
-    if (lockedModStats) {
-      _.forIn(lockedModStats, (value, statHash) => {
-        baseStats[statHash] += value;
-      });
-    }
-  }
-  // mapping out from stat values to ensure ordering and that values don't fall below 0 from locked mods
-  return statValues.map((statHash) => Math.max(baseStats[statHash], 0));
+  // mapping out from stat values to ensure ordering
+  return statValues.map((statHash) => keyedStats[statHash]);
```

Comment:
I'm not sure why you need to construct a map and then turn it into an array instead of just constructing an array directly.

Reply:
It was just to ensure ordering of the stats but now that I think of it maybe we need to use the stat filter order...
I will try to figure out what is more correct tomorrow.

Reply:
Ok so upon investigation this needs to match the stat ordering of the ArmorSet.firstValidSetStatChoices. I will update it to use this.